File: aoc-dec01/dec01.jl
testdata = Parse.(Int,split(read(joinpath(@__DIR__,"input.txt"),String)));

dec01_1(x) = first(i*j for i ∈ x, j ∈ x if i+j == 2020)

dec01_2(x) = first(i*j*k for i ∈ x, j ∈ x, k ∈ x if i+j+k == 2020)

using Test
@test dec01_1(testdata) == 259716
@test dec01_2(testdata) == 120637440

File: aoc-dec01/dec01.clj
(def x [1721
        979
        366
        299
        675
        1456])

(take 1 (for [i  x j x :when (= (+ i j) 2020)] (* i j)))

(require '[clojure.string :as str])

(def testdata (slurp "../aoc-dec01/input.txt"))

(defn dec01-1 [inp]
  (def x (str/split-lines inp))
  (take 1 (for [i x j x :when (= (+ i j) 2020)] (* i j))))

(dec01-1 testdata)

(defn dec01-2 [inp]
  (def x (str/split-lines inp))
  (take 1 (for [i x j x k x :when (= (+ i j k) 2020)] (* i j k))))

(dec01-2 testdata)

(str/split-lines testdata)